lib/collection/response.js

```diff
@@ -189,8 +196,9 @@ _.inherit((
 
 _.assign(Response.prototype, /** @lends Response.prototype */ {
     update: function (options) {
-        // options.stream accepts new Buffer() as well as new Buffer().toJSON()
-        var stream = normaliseStream(options.stream);
+        // options.stream accepts Buffer, Buffer.toJSON() or base64 string
+        // @todo this doubles the memory footprint (options.stream + generated buffer).
```

Comment:
Do you mean it doubles the footprint or you mean that it causes temporary consumption spike?

Reply:
Yes, causes temporary consumption spike.